### DISCUSSION TITLE
Nested dynamic Field-Arrays

### DISCUSSION THREAD

**Discussion by Kay2dan** [react-hook-form/react-hook-form#5884] [Q&A] [answered]
Hi, 

Is there a guide on how to set nested field-arrays.

I have data in the form:

```javascript
const data = [
  {
    a  'aaa1',
    b: 'bbb1',
    c: [
      {
        nestedD : 'ddd1',
        nestedE: 'eee1',
        ...
      }, 
      ...
    ]
  }
]
```
To summarise above data-struct: I have nested array of input fields within an array of fields. The user will also append more nested input fields as per their needs (within `c `above, or other child objects of data array).

At the moment my code looks as such:

```javascript

const GoalsForm: FC<GoalsFormFCPropTypes> = ({...}) => {
  // ... setting up various states
  const ctx = useForm(); // initiate form
  const { fields, append, remove } = useFieldArray({
    control: ctx.control,
    name: "goals",
  });

  useEffect(() => {
  // ... data manipulation to format for the form struct 
  const formatted = dataFromBackend.map((d, i) => ... );
  ctx.reset({ goals: formatted });
  }, [dataFromBackend]);

  return(
    <div>
      {fields.map((item, i) => {
        return(
           <GoalForm ...
        );
      )}
    </div>

```

I am not clear how to instantiate the child `useFieldArray` (for the prop `c` in the above object) after the data is formatted.

**Comment by bluebill1049** [comment, marked as answer]
> To summarise above data-struct: I have nested array of input fields within an array of fields. The user will also append more nested input fields as per their needs (within c above, or other child objects of data array).

you should be able to append to the nested array as well.

but if you need to update a single node with nested value, you will have to use `setValue` until the `update` method gets released in the coming weeks.

**Reply by Kay2dan** [reply]
Thank you for getting back so quickly Bill.

At first, I was thinking, he should have elaborated on his statement. Then, I thought about testing my assumptions and lo, it worked seamlessly. 

RHF is such a thoughtfully designed library. Im a fan.

Thanks again.
